lil-lab/recnet
# [recnet-web] RecNet Slack OAuth Flow

## Description


Finish slack oauth flow.

## Related Issue






- #261 
- #64 

## Notes




## Test


Before test, update `.env.local`
- `NEXT_PUBLIC_BASE_URL` should be `https://localhost:3000`
- Add `SLACK_APP_CLIENT_ID`
- Add `SLACK_OAUTH_APP_SCOPES` 
- Add`SLACK_OAUTH_REDIRECT_URI="/api/slack/oauth/callback"`

Run `nx dev:ssl recnet` and go to `https://localhost:3000`
- Go to setting and subscription tab -> Click the button to install slack app to your workspace
- Should redirect you to slack page
- Should able to choose different workspace
- After clicking "Allow", should redirect back to frontend and see the success message
- Go to setting again, should see the workspace name of where you install the app


## Screenshots (if appropriate):


![Screenshot 2024-11-21 at 5 59 40 PM](https://github.com/user-attachments/assets/76d381f6-4dfd-49ee-976a-944b686a6d55)
![Screenshot 2024-11-21 at 5 59 50 PM](https://github.com/user-attachments/assets/092f53a1-81aa-4914-bebb-2ebc6f27f1d5)
![Screenshot 2024-11-21 at 6 00 07 PM](https://github.com/user-attachments/assets/cae362f9-58b8-4b0a-9f2a-ea4c1eb7dfc4)

## TODO
